### PR TITLE
Ignore UTS46 validity criteria V2

### LINKED ISSUE #160: Consider ignoring UTS46 validity criteria V2
```
Each of the following criteria must be satisfied for a label:
2. The label must not contain a U+002D HYPHEN-MINUS character in both the third and fourth positions. 
```

However, it seems that other UAs don't enforce this, and it matches domains commonly used on youtube. ex: r6---sn-5hne6nee.googlevideo.com

### PR DESCRIPTION
**Pull request by canaltinova**
Fixes #160
r? @SimonSapin or @Manishearth

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-url/240)

### CODE REVIEW

File: idna/src/uts46.rs [hunk 1]
```diff
@@ -198,12 +198,10 @@ fn validate(label: &str, flags: Flags, errors: &mut Vec<Error>) {
...
 
     // Can not contain '.' since the input is from .split('.')
-    if {
-        let mut chars = label.chars().skip(2);
-        let third = chars.next();
-        let fourth = chars.next();
-        (third, fourth) == (Some('-'), Some('-'))
-    } || label.starts_with("-")
+    // Spec says, the label must not contain a HYPHEN-MINUS character in both the
+    // third and fourth positions. But nobody follow this criteria. See the spec issue below:
```
Comment (Manishearth) [inline idna/src/uts46.rs]: "Spec says that the", and "follows"

**Comment by Manishearth** [comment]
LGTM but I would like sign-off from @SimonSapin here

File: idna/src/uts46.rs (hunk 1 above)
Comment (canaltinova) [inline idna/src/uts46.rs]: Fixed, thanks!

File: idna/src/uts46.rs [hunk 2]
```diff
@@ -198,12 +198,10 @@ fn validate(label: &str, flags: Flags, errors: &mut Vec<Error>) {
     }
 
     // Can not contain '.' since the input is from .split('.')
-    if {
-        let mut chars = label.chars().skip(2);
-        let third = chars.next();
-        let fourth = chars.next();
-        (third, fourth) == (Some('-'), Some('-'))
-    } || label.starts_with("-")
+    // Spec says that the, label must not contain a HYPHEN-MINUS character in both the
```
Comment (Manishearth) [inline idna/src/uts46.rs]: no comma

**Comment by emilio** [comment]
Isn't this the same as #208? If so, it's really sad that that PR has been sitting there for so long :(

**Comment by Manishearth** [comment]
It removes the configurability.

**Comment by canaltinova** [comment]
Fixed the typo

**Comment by SimonSapin** [comment]
@bors-servo r+

Thanks!

**Pull request merged**